.github/workflows/post-pull-request-checks-automation.yml

```diff
@@ -40,26 +40,43 @@ on:
         type: number
         default: 1
         description: Configuration for the timeout-minutes parameter of the workflow.
+      run-only-latest-workflow:
+        type: string
+        description: Run only if the current workflow is the most recent workflow.
+        default: true
 
 jobs:
   post-pull-request-checks-automation:
     runs-on: ubuntu-22.04
     timeout-minutes: ${{ inputs.timeout-minutes }}
     steps:
+      - name: Check Further Steps Should Run
+        uses: Lombiq/GitHub-Actions/.github/actions/check-current-workflow-is-latest@issue/OSOE-655
+        id: check-steps-should-run
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          current-workflow-id: ${{ github.run_id }}
+        if: inputs.run-only-latest-workflow == 'true'
+
       - name: Set Merge Token
         shell: pwsh
         env:
           MERGE_TOKEN: ${{ secrets.MERGE_TOKEN }}
         run: |
           $mergeToken = $Env:MERGE_TOKEN ? $Env:MERGE_TOKEN : "${{ secrets.GITHUB_TOKEN }}"
           "MERGE_TOKEN=$mergeToken" >> $Env:GITHUB_ENV
+        if: |
+          (inputs.run-only-latest-workflow == 'false') || (steps.check-steps-should-run.outputs.should-run)
```

Comment:
Is the `inputs.run-only-latest-workflow == 'false'` part needed here? Since `steps.check-steps-should-run.outputs.should-run` won't be set anyway if that step hasn't run. Also, you need `steps.check-steps-should-run.outputs.should-run == 'true'`.